### PR TITLE
Added husky and prepush tests.

### PR DESCRIPTION
**Pull request by lucasconstantino**
This is a suggestion only. Added husky package for automated githooks. This will force tests passing on the "prepush" hook for any contributing user.

Ps.: `git push --no-verify` would skip the test, if anybody get stuck at something.

### CODE REVIEW

**Comment by stubailo** [comment]
Hmmm, seems like this will actively discourage people to open PRs with failing tests, which is something we actually want right?

I think having linting and stuff like that on hooks makes sense, but not tests IMO.

File: package.json [hunk 1]
```diff
@@ -79,6 +80,7 @@
     "grunt": "1.0.1",
     "grunt-tslint": "^5.0.1",
     "gzip-size": "^3.0.0",
+    "husky": "^0.14.3",
```
Comment (stubailo) [inline package.json]: What does `husky` do?

**Comment by lucasconstantino** [comment]
I see your point. I guess I'm too used to having these hooks around that I got comfortable with people in my team making good judgement on whether or not forcing a push with failing tests. This has highly increase test coverage, though, for our projects. But these projects are no lib no do we use pull-requests, thus we don't rely on strong code reviewing processes - what Apollo seems to have. Having that in mind, maybe you don't really need some automation process avoiding non-passing pushes.

About **husky**, it is a tool which after `npm install` will automatically install git-hook based on scripts present on your `package.json`.

**Comment by stubailo** [comment]
Yeah, we actually disabled pushing to `master` completely so things can only be merged via PR - and any PRs that don't pass tests can't be merged. So in this particular repository it doesn't feel like we need another thing verifying test running, but I'm curious what @jbaxleyiii thinks.

**Comment by lucasconstantino** [comment]
Honestly I think you're completely right. I'll close this one... sorry for the noise! We can reopen should anyone have other arguments in favor.

**Comment by stubailo** [comment]
No worries, great discussion anyway!

**Pull request closed without merging**